src/faction_camp.cpp

```diff
@@ -535,23 +535,38 @@ static cata::optional<basecamp *> get_basecamp( npc &p, const std::string &camp_
 recipe_id base_camps::select_camp_option( const std::map<recipe_id, translation> &pos_options,
         const std::string &option )
 {
-    std::vector<recipe_id> pos_name_ids;
     std::vector<std::string> pos_names;
+    int choice = 0;
+
     for( const auto &it : pos_options ) {
         pos_names.push_back( it.second.translated() );
-        pos_name_ids.push_back( it.first );
     }
 
-    if( pos_name_ids.size() == 1 ) {
-        return pos_name_ids.front();
+
+    if( pos_options.size() != 1 ) {
+        for( size_t i = 0; i < pos_names.size() - 1; i++ ) {
+            for( size_t k = i + 1; k < pos_names.size(); k++ ) {
+                if( localized_compare( pos_names[k], pos_names[i] ) ) {
+                    std::swap( pos_names[i], pos_names[k] );
+                }
+            }
+        }
```

Comment:
This was the payoff for refactoring, using std::sort instead of a hand-coded sort.
```suggestion
        std::sort( pos_names.begin(), pos_names.end(), localized_compare );
```

Reply:
You can't deplete the list of expansions, since you can build the same one in multiple locations. However, you can try to build one in a location that has no choices. If I try a road I get a response that I can't build in a road_un, generated from somewhere before this operation. I haven't tested a swamp, though (legal in principle, but no such expansions exist currently). Tested that as well: "You can't build any expansions in a forest." Thus, code further up the chain bails out before reaching this operation on an empty list (it can also be noted that the code handling of that case is unchanged: it would feed an empty list into uilist, which I would guess you return with a negative value).